app/upgrades/v16/upgrades.go

```diff
@@ -87,6 +88,13 @@ func CreateUpgradeHandler(
 		} else if addErr != nil {
 			return vm, errorsmod.Wrapf(addErr, "unable to add rate limits")
 		}
+
+		// Set CosmWasm params
+		wasmParams := wasmtypes.DefaultParams()
+		wasmParams.CodeUploadAccess = wasmtypes.AllowNobody
+		// TODO(reece): only allow specific addresses to instantiate contracts or anyone with AccessTypeEverybody?
+		wasmParams.InstantiateDefaultPermission = wasmtypes.AccessTypeAnyOfAddresses
```

Comment:
need to review this is correct & handle TODO here.

Reply:
We want the governance to handle both code upload and instantiation. 

IIUC, to use permissiond wasm we would need to use these settings:
```golang
wasmParams := wasmtypes.Params{
	CodeUploadAccess:             wasmtypes.AllowNobody,
	InstantiateDefaultPermission: wasmtypes.AllowNobody,
}
```

I don't think that code upload is followed by an instantiation operation by default. But, that can be can be achieved through a gov prop that has 2 Txs to execute: `[MsgStoreCode, MsgInstantiateContract]`.

Do you have any comments on this @Reecepbcups ?

Reply:
This makes total sense yes, wasmtypes.AllowNobody is the desired param for both

Reply:
I think I'm getting it, thanks!

We will address those concerns before testnet and write new docs before cosmwasm lands on the hub mainnet.

Reply:
Hi! Do you have an example of what a proposal that uses `MsgStoreAndInstantiateContract` would look like? I would like to know how the `wasm_byte_code` field must be encoded:
```
{
 "messages": [
  {
   "@type": "/cosmwasm.wasm.v1.MsgStoreAndInstantiateContract",
   "authority": "cosmos10d07y265gmmuvt4z0w9aw880jnsr700j6zn9kn",
   "wasm_byte_code": null,  <- My understanding is the .wasm file contents will go in here
   "instantiate_permission": null,
   "unpin_code": false,
   "admin": null,
   "label": "label",
   "msg": null,
   "funds": [],
   "source": null,
   "builder": null,
   "code_hash": null
  }
 ],
 "metadata": "ipfs://CID",
 "deposit": "10000000uatom",
 "title": "Test",
 "summary": "no"
}
```

Reply:
@dasanchez Yes, it can be raw or gzip compressed. Base64 encoded

Reply:
@Reecepbcups thanks! I ended up using the `submit-proposal` command from the wasm module (rather than the gov draft-proposal template) and managed to 1) store and instantiate, and 2) execute via proposals:
```
gaiad tx wasm submit-proposal store-instantiate my_first_contract.wasm '{"count":100}' \
    --label "my first contract" \
    --no-admin \
    --instantiate-nobody true \
    --title "Store and instantiate CW template" \
    --summary "This proposal will store and instantiate the cw template contract"
```

```
gaiad tx wasm submit-proposal execute-contract $contract_address '{"increment":{}}' --title "Execute CW template" --summary "This proposal will execute an increment on the cw template contract"
```

```
gaiad q wasm contract-state smart $contract_address '{"get_count":{}}'
data:
  count: 101
```